### PR TITLE
Fix/3672 private pages api

### PR DESCRIPTION
**Pull request by NabilMostafa**
Fix issue #[3672](https://github.com/wagtail/wagtail/issues/3672)
# Changes 

- [x]  If user is not authenticated, return public pages only
- [x]  If user is authenticated, return pages that has login restriction. Also, check if there are any group restriction and check if user belongs to any of the groups in the page restriction groups
- [x] Cover changes in tests

### CODE REVIEW

File: wagtail/api/v2/views.py [hunk 1]
```diff
@@ -440,8 +441,16 @@ def get_base_queryset(self):
...
-        queryset = Page.objects.all().public().live()
+        # Get all live pages
+        queryset = Page.objects.all().live()
+        # if user is not authenticated return only public pages
+        if not self.request.user.is_authenticated:
+            queryset &= queryset.public()
+        else:  # get pages that are accessible to logged-in users or users in specific groups
+            queryset &= queryset.filter(Q(view_restrictions__restriction_type=PageViewRestriction.LOGIN)
+                                        | Q(view_restrictions__groups__in=self.request.user.groups.all())
+                                        | Q(view_restrictions__restriction_type=None))
```
Comment (RealOrangeOne) [inline wagtail/api/v2/views.py:452]: Indentation here looks a little off
```suggestion
            queryset &= queryset.filter(
                Q(view_restrictions__restriction_type=PageViewRestriction.LOGIN) |
                Q(view_restrictions__groups__in=self.request.user.groups.all()) |
                Q(view_restrictions__restriction_type=None)
            )
```

Comment (NabilMostafa) [reply wagtail/api/v2/views.py:452]: You're right, but your suggestion broke this rule https://www.flake8rules.com/rules/W504.html, I think it should be like that ->
```suggestion
          queryset &= queryset.filter(
                Q(view_restrictions__restriction_type=PageViewRestriction.LOGIN)
                | Q(view_restrictions__groups__in=self.request.user.groups.all())
                | Q(view_restrictions__restriction_type=None)
            )
```

File: wagtail/api/v2/tests/test_pages.py [hunk 2]
```diff
@@ -81,6 +84,48 @@ def test_private_pages_dont_appear_in_list(self):
         content = json.loads(response.content.decode('UTF-8'))
         self.assertEqual(content['meta']['total_count'], new_total_count)
 
+    def test_private_pages_with_user_login(self):
+        client = APIClient()
```
Comment (kaedroho) [inline wagtail/api/v2/tests/test_pages.py:88]: Any reason to not use ``self.client`` here?

**Review by kaedroho — CHANGES_REQUESTED** [review]

File: wagtail/api/v2/tests/test_pages.py (hunk 2 above)
Comment (NabilMostafa) [reply wagtail/api/v2/tests/test_pages.py:88]: For some reason when I used `self.client` and `self.login` the user wasn't logged in the views and always returned `anonymous` user type

**Comment by lb-** [comment]
@kaedroho  just a ping to see if you can re-review this, looks like changes requested have been addressed

**Review by kaedroho — APPROVED** [review]
This looks good to me, thanks!

**Pull request merged**